### ISSUE TITLE
TimeoutException when Jetty server stops (3.0.0-beta-2)

### ISSUE THREAD

**Issue by martin-tarjanyi** [wiremock/wiremock#2073]
# System info
**Version**: 3.0.0-beta-2
**OS**: Windows 10 21H2
**Java**: 17
**Spring Boot**: 3.0.2

# Repro
Wiremock is running as part of a Spring integration test using WiremockExtension:
```java
@RestClientTest
@AutoConfigureMockRestServiceServer(enabled = false)
public class BaseHttpClientTest {
    @RegisterExtension
    protected static WireMockExtension wireMockServer = WireMockExtension.newInstance()
        .options(wireMockConfig().dynamicPort())
        .build();

   // test cases...
}
```

When the tests finish the following exception is raised:
```
java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at org.eclipse.jetty.server.Server.doStop(Server.java:490)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:132)
	at com.github.tomakehurst.wiremock.jetty.JettyHttpServer.stop(JettyHttpServer.java:243)
	at com.github.tomakehurst.wiremock.WireMockServer.stop(WireMockServer.java:147)
	at com.github.tomakehurst.wiremock.junit5.WireMockExtension.stopServerIfRunning(WireMockExtension.java:196)
	at com.github.tomakehurst.wiremock.junit5.WireMockExtension.afterAll(WireMockExtension.java:232)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeAfterAllCallbacks$18(ClassBasedTestDescriptor.java:461)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeAfterAllCallbacks$19(ClassBasedTestDescriptor.java:461)
```

# Workaround
## Set higher stop timeout
```
.options(wireMockConfig().dynamicPort().jettyStopTimeout(10_000L))
```
Jetty server takes ~8-9 seconds to stop in this case.
These are the produced logs at stopping:
```
2023-01-28T16:21:37.992+01:00  INFO 8408 --- [    Test worker] org.eclipse.jetty.server.Server          : Stopped Server@1885905b{STOPPING}[11.0.13,sto=10000]
2023-01-28T16:21:37.992+01:00  INFO 8408 --- [    Test worker] org.eclipse.jetty.server.Server          : Shutdown Server@1885905b{STOPPING}[11.0.13,sto=10000]
2023-01-28T16:21:45.939+01:00  INFO 8408 --- [    Test worker] o.e.jetty.server.AbstractConnector       : Stopped NetworkTrafficServerConnector@43423d40{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:0}
2023-01-28T16:21:45.945+01:00  INFO 8408 --- [    Test worker] o.e.jetty.server.handler.ContextHandler  : Stopped o.e.j.s.ServletContextHandler@7602c65d{/,null,STOPPED}
2023-01-28T16:21:45.946+01:00  INFO 8408 --- [    Test worker] o.e.jetty.server.handler.ContextHandler  : Stopped o.e.j.s.ServletContextHandler@1165a952{/__admin,null,STOPPED}
```

## Set zero (no) timeout
```
.options(wireMockConfig().dynamicPort().jettyStopTimeout(0L))
```
Interestingly, Jetty server takes only ~2-3 seconds to stop in this case.
Logs are slightly different in this case
```
2023-01-28T16:23:34.558+01:00  INFO 3404 --- [    Test worker] org.eclipse.jetty.server.Server          : Stopped Server@1885905b{STOPPING}[11.0.13,sto=0]
2023-01-28T16:23:34.564+01:00  INFO 3404 --- [    Test worker] o.e.jetty.server.AbstractConnector       : Stopped NetworkTrafficServerConnector@43423d40{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:0}
2023-01-28T16:23:34.566+01:00  INFO 3404 --- [    Test worker] o.e.jetty.server.handler.ContextHandler  : Stopped o.e.j.s.ServletContextHandler@7602c65d{/,null,STOPPED}
2023-01-28T16:23:34.566+01:00  INFO 3404 --- [    Test worker] o.e.jetty.server.handler.ContextHandler  : Stopped o.e.j.s.ServletContextHandler@1165a952{/__admin,null,STOPPED}
2023-01-28T16:23:37.066+01:00  WARN 3404 --- [qtp775739186-21] o.e.jetty.util.SharedBlockingCallback    : Blocker not complete Blocker@f274a72{null}
```

# Expected behavior
* Jetty server stops without error on default options
* Jetty server stops in a timely manner, on 2.35.0 server stopping is below 100ms

**Comment by OmarHawk** [comment]
Having the same issue - both workarounds do work for us (also when manually initializing Wiremock object like this):

```
new WireMockServer(
            wireMockConfig().port(SERVER_PORT).jettyStopTimeout(0L));
```

instead of like this:
```
new WireMockServer(SERVER_PORT);
```

But looking not so nice... :/

**Comment by OmarHawk** [comment]
Fixed with 3.0.0-beta-4 (at least for me it works now again)

**Comment by tomakehurst** [comment]
Yep, beta-4 improved Jetty shutdown by explicitly closing all of the TCP connections before stopping so this issue should be avoided now.
